### PR TITLE
Tutorial "Plotting text": Rewrite to improve structure, add parameters, add list input

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to expand and update the tutorial "Plotting text":
- [x] Mention the `position` parameter along with `justify`
- [x] Mention the `pen` parameter along with `fill`
- [x] Grouping and showing the configurations  `justify`,  `font`, `angle`, `fill` + `pen` via simple Cartesian plots 
- [x] Showing the possiblity of providing lists for `justify`, `font`, and `angle` (possible after PR #2720) beside providing an external input file

Please state, whether these changes are improvements, or that we should keep the current version.

Changes suggested by @seisman to improve the structure of the tutorial (see comment https://github.com/GenericMappingTools/pygmt/pull/2760#issuecomment-1780632322):
- [x] 1. The first map should have only one text (e.g., `text="SOUTH CHINA SEA", x=112, y=6`)
- [x] 2. Introduce the `angle`/`font`/`justify`/`pen`/`fill` parameters to adjust the texts (maybe one map for `angle`/`font`/`justify` and another map for `pen`/`fill`
- [x] 3. Show how to add several texts in one `Figure.text` call (e.g., pass 1-D arrays to `x`/`y`/`text`)
- [x] 4. Show that `angle`/`font`/`justify` can also accept 1-D arrays (maybe combine 3 and 4 into one map)
- [ ] 5. Explain that it's also possible to pass a file or a 2-D matrix to ~`data`~ `textfiles`
- [x] 6. Introduce the `position` parameter. A useful use case for this parameter is to add a tag  (e.g., `(a)`) at the top left corner of the map for multiple subplots, but we should also mention that the `Figure.subplot` method can do this automatically.

**Preview**: https://pygmt-dev--2760.org.readthedocs.build/en/2760/tutorials/basics/text.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Sorry to be mean here, but I think the "Plotting text" tutorial (both the old version and the current new version) is not as good as other tutorials. In other tutorials, we usually start with a simple plot with a few required parameters and then improve the plot by adding more parameters step by step in the following sections, but in this tutorial, we start with a nice map and then start to explain the `position`/`text`, `font`, `angle`, `fill`/`pen` parameters, which has nothing to do with the first map. The tutorial reads more like a manual for `Figure.text`'s parameters, rather than a tutorial for adding texts.

Here are my thoughts about the tutorial:

1. The first map should have only one text (e.g., `text="SOUTH CHINA SEA", x=112, y=6`)
2. Introduce the `angle`/`font`/`justify`/`pen`/`fill` parameters to adjust the texts (maybe one map for `angle`/`font`/`justify` and another map for `pen`/`fill`
3. Show how to add several texts in one `Figure.text` call (e.g., pass 1-D arrays to `x`/`y`/`text`)
4. Show that `angle`/`font`/`justify` can also accept 1-D arrays (maybe combine 3 and 4 into one map)
5. Explain that it's also possible to pass a file or a 2-D matrix to `data`
6. Introduce the `position` parameter. A useful use case for this parameter is to add a tag  (e.g., `(a)`) at the top left corner of the map for multiple subplots, but we should also mention that the `Figure.subplot` method can do this automatically.

**Comment by yvonnefroehlich** [comment]
> Sorry to be mean here, but I think the "Plotting text" tutorial (both the old version and the current new version) is not as good as other tutorials. In other tutorials, we usually start with a simple plot with a few required parameters and then improve the plot by adding more parameters step by step in the following sections, but in this tutorial, we start with a nice map and then start to explain the `position`/`text`, `font`, `angle`, `fill`/`pen` parameters, which has nothing to do with the first map. The tutorial reads more like a manual for `Figure.text`'s parameters, rather than a tutorial for adding texts.
> 
> Here are my thoughts about the tutorial:
> 
> 1. The first map should have only one text (e.g., `text="SOUTH CHINA SEA", x=112, y=6`)
> 2. Introduce the `angle`/`font`/`justify`/`pen`/`fill` parameters to adjust the texts (maybe one map for `angle`/`font`/`justify` and another map for `pen`/`fill`
> 3. Show how to add several texts in one `Figure.text` call (e.g., pass 1-D arrays to `x`/`y`/`text`)
> 4. Show that `angle`/`font`/`justify` can also accept 1-D arrays (maybe combine 3 and 4 into one map)
> 5. Explain that it's also possible to pass a file or a 2-D matrix to `data`
> 6. Introduce the `position` parameter. A useful use case for this parameter is to add a tag  (e.g., `(a)`) at the top left corner of the map for multiple subplots, but we should also mention that the `Figure.subplot` method can do this automatically.

No worries @seisman. My initial intent for this PR was only to add a code example regarding list input for `font`, `angle`, and `justify`. But this turned out to be more difficult than expected because of the current structure of the tutorial. Thus, I made some additional changes to the tutorial and then tried to include the list-input-aspect. But as you mentioned, the tutorial is still not optimal in terms of its structure from an educational perspective. However, I was unsure about making more significant changes, as I am not the original author of this tutorial. But I am open to consider your suggested points above to restructure and rewrite this tutorial. Maybe it makes sense to see what the other @GenericMappingTools/pygmt-maintainers think about these changes🙂 before I start working on this.

**Comment by yvonnefroehlich** [comment]
There is again an issue with building the documentation:
```
generating gallery...
generating gallery for gallery... 
[2Kgenerating gallery for gallery/maps... [ 17%] choropleth_map.py
Segmentation fault (core dumped)
```
After removing `chorophlet_map.py`:
```
[2Kgenerating gallery for gallery/lines... [ 85%] roads.py
Segmentation fault (core dumped)
Command time: 39s Return: 139
```
After also removing `roads.py`, building the documentation is successful.

**Comment by yvonnefroehlich** [comment]
I tried to address the issues mentioned by @seisman in https://github.com/GenericMappingTools/pygmt/pull/2760#issuecomment-1780632322. @GenericMappingTools/pygmt-maintainers think current changes are improving this tutorial?

File: examples/tutorials/basics/text.py [hunk 1]
```diff
@@ -12,146 +12,233 @@
...
 #
-# Text annotations can be added to a map using the :meth:`pygmt.Figure.text`
-# method of the :class:`pygmt.Figure` class.
-#
-# Here we create a simple map and add an annotation using the ``text``, ``x``,
-# and ``y`` parameters to specify the annotation text and position in the
-# projection frame. ``text`` accepts *str* types, while ``x`` and ``y``
-# accept either *int* or *float* numbers, or a list/array of numbers.
+# To add a single text label to a plot the ``text`` and ``x`` and ``y`` parameters to
+# specify the text and position.
```
Comment (seisman) [inline examples/tutorials/basics/text.py:19]: This sentence doesn't read well. Missing a verb?
```suggestion
# To add a single text label to a plot, use the ``text`` and ``x`` and ``y`` parameters to
# specify the text and position.
```

File: examples/tutorials/basics/text.py [hunk 2]
```diff
@@ -12,146 +12,233 @@
...
 
 fig = pygmt.Figure()
-with pygmt.config(MAP_FRAME_TYPE="plain"):
-    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
-fig.coast(land="black", water="skyblue")
-
-# Plot text annotations using single arguments
-fig.text(text="SOUTH CHINA SEA", x=112, y=6)
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
```
Comment (seisman) [inline examples/tutorials/basics/text.py:23]:
```suggestion
```

File: examples/tutorials/basics/text.py (hunk 1 above)
Comment (yvonnefroehlich) [reply examples/tutorials/basics/text.py]: Oh, yes, thanks for spotting!

File: examples/tutorials/basics/text.py [hunk 3]
```diff
@@ -12,146 +12,232 @@
...
-    fig.basemap(region=[108, 120, -5, 8], projection="M20c", frame="a")
-fig.coast(land="black", water="skyblue")
-
-# Plot text annotations using single arguments
-fig.text(text="SOUTH CHINA SEA", x=112, y=6)
-
-# Plot text annotations using lists of arguments
-fig.text(text=["CELEBES SEA", "JAVA SEA"], x=[119, 112], y=[3.25, -4.6])
+fig.basemap(region=[-5, 5, -5, 5], projection="X5c", frame=True)
+fig.text(x=0, y=0, text="My text")
```
Comment (seisman) [inline examples/tutorials/basics/text.py:24]:
```suggestion
```

**Review by seisman — COMMENTED** [review]
Looks good.

**Review by seisman — APPROVED** [review]

**Pull request merged**